content/shmem_team_collect.tex

```diff
@@ -0,0 +1,96 @@
+\apisummary{
+    Concatenates blocks of data from multiple \acp{PE} in a team to an array in every
+    \ac{PE} in the team.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void @\FuncDecl{shmem\_team\_collect32}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+void @\FuncDecl{shmem\_team\_collect64}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+void @\FuncDecl{shmem\_team\_fcollect32}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+void @\FuncDecl{shmem\_team\_fcollect64}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+\end{Csynopsis}
+
+\begin{apiarguments}
+
+\apiargument{IN}{team}{A valid \openshmem team handle to a team which has been
+    created without disabling support for collective operations.}
+\apiargument{OUT}{dest}{A symmetric array large enough
+    to accept the concatenation of the \source{} arrays on all \acp{PE} in the team.
+    See table below in this description for allowable data types.}
+\apiargument{IN}{source}{A symmetric data object that can be of any type permissible
+    for the \dest{} argument.}
+\apiargument{IN}{nelems}{The number of elements in the \source{} array. \VAR{nelems}
+    must be of type \VAR{size\_t}.}
+
+\end{apiarguments}
+
+\apidescription{   
+    \openshmem \FUNC{team\_collect} and \FUNC{team\_fcollect} are collective routines
+    over an existing team. These routines concatenate \VAR{nelems}
+    \CONST{64}-bit or \CONST{32}-bit data items from the \source{} array into the
+    \dest{} array, over all \acp{PE} in the specified \VAR{team} in processor number order.
+    The resultant \dest{} array contains the contribution from the first \ac{PE} in the \VAR{team}
+    , then the contribution from the second \ac{PE} in the \VAR{team}, and so on.
+    The collected result is written to the \dest{} array for all \acp{PE} in the team.
+    
+    The \FUNC{fcollect} routines require that \VAR{nelems} be the same value in all
+    \acp{PE} in the team, while the \FUNC{collect} routines allow \VAR{nelems} to
+    vary from \ac{PE} to \ac{PE}.
+
+    As with all \openshmem team collective routines, each of these routines assumes that
+    only \acp{PE} in the given team call the routine.  If a \ac{PE} not in the
+    team calls an \openshmem team collective routine, the behavior is undefined.
```

Comment:
We can remove this section based on how we define team handles. Team handles are only valid on the PE that participated in the split operation that created the team, so it is not possible for a PE not in the team to call the collective routine except with some invalid handle.